### PR TITLE
Logs of a sync

### PR DESCRIPTION
**Pull request by german-mergestat**
Implementing pages (#70):

- Syncs list with its logs

logs-sync

- Logs detail of a specific sync

log-detail

### CODE REVIEW

File: ui/src/api-logic/mappers/syncs-logs.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        records: q.repoSyncLogs.totalCount,
+        duration: q?.doneAt ? getSimpleDurationTime(new Date(q?.startedAt), new Date(q?.doneAt)) : 'running',
+        syncStart: q?.startedAt,
+      }
+
+      // 3. Get list logs info 2021/06/09 02:21 This is a log line ...
+      const logs: Array<string> = []
+      q.repoSyncLogs.nodes.forEach((log) => {
+        logs.push(`${lightFormat(new Date(log.createdAt), DATE_FORMAT)} ${log.logType}: ${log.message} ...`)
+      })
```
Comment (patrickdevivo) [inline ui/src/api-logic/mappers/syncs-logs.ts:49]: Could a call to `.map(...)` here be a little more succinct than using `forEach`?

File: ui/src/utils/constants.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const DATE_FORMAT = 'yyyy/MM/dd HH:mm'
+export const GITHUB_URL = 'https://github.com/'
```
Comment (patrickdevivo) [inline ui/src/utils/constants.ts:2]:
```suggestion
export const GITHUB_URL = 'https://github.com/'

```

File: ui/src/utils/index.tsx [hunk 3]
```diff
@@ -34,6 +35,11 @@ export function selectAllState(data: Array<any>) {
       : CHECKBOX_STATES.Indeterminate
 }
 
+/**
+ * Method to get time ago respect current date
```
Comment (patrickdevivo) [inline ui/src/utils/index.tsx:39]:
```suggestion
 * Method to get time ago relative to current date
```

**Comment by patrickdevivo** [comment]
FYI I'm seeing some odd values for `Latest Run` - I'm guessing there's a funky date value being passed in somewhere? This should show the relative time ago of the latest _completed_ run (which will have a `done_at` field)

image

**Comment by patrickdevivo** [comment]
In the breadcrumb nav, for the last item, let's show the timestamp of the sync vs the status? And use an icon to indicate status

image

File: ui/src/utils/index.tsx (hunk 3 above)
Comment (german-mergestat) [reply ui/src/utils/index.tsx]: Done [eb2c514](https://github.com/mergestat/fuse/pull/118/commits/eb2c5142974b33c084a05c91150ed0888c3b9e1c)

File: ui/src/utils/constants.ts (hunk 2 above)
Comment (german-mergestat) [reply ui/src/utils/constants.ts]: Missed again XD. Done [eb2c514](https://github.com/mergestat/fuse/pull/118/commits/eb2c5142974b33c084a05c91150ed0888c3b9e1c)

File: ui/src/api-logic/mappers/syncs-logs.ts (hunk 1 above)
Comment (german-mergestat) [reply ui/src/api-logic/mappers/syncs-logs.ts]: mmm good catch, yep. Done [eb2c514](https://github.com/mergestat/fuse/pull/118/commits/eb2c5142974b33c084a05c91150ed0888c3b9e1c)

File: ui/src/api-logic/mappers/syncs-logs.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  // 1. Get Sync general info
+  data?.repo?.repoSyncs.nodes.forEach((s) => {
+    const syncInfo = {
+      id: s.id,
+      title: s?.syncType.replaceAll('_', ' ') || '',
+      brief: s?.repoSyncTypeBySyncType?.description || '',
+      syncState: s?.repoSyncQueues.nodes.length !== 0 ? mapToRepoSyncStateT(s?.repoSyncQueues.nodes[0]?.status || '') : 'empty',
+    }
+    repoData.sync = syncInfo
```
Comment (patrickdevivo) [inline ui/src/api-logic/mappers/syncs-logs.ts:30]:
```suggestion
    repoData.sync = {
      id: s.id,
      title: s?.syncType.replaceAll('_', ' ') || '',
      brief: s?.repoSyncTypeBySyncType?.description || '',
      syncState: s?.repoSyncQueues.nodes.length !== 0 ? mapToRepoSyncStateT(s?.repoSyncQueues.nodes[0]?.status || '') : 'empty',
    }
```

**Review by patrickdevivo — DISMISSED** [review]

File: ui/src/api-logic/mappers/syncs-logs.ts (hunk 4 above)
Comment (german-mergestat) [reply ui/src/api-logic/mappers/syncs-logs.ts:30]: I'm gonna merge this branch, and gonna fix this point in next PR :)

Comment (german-mergestat) [reply ui/src/api-logic/mappers/syncs-logs.ts]: Done [cf5ddad](https://github.com/mergestat/fuse/pull/118/commits/cf5ddadcfbec46a96bea8a2e89f00252cdd0f448)

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**